### PR TITLE
cmake: add version 3.16.8

### PR DESCRIPTION
**Pull request by ledocc**
Specify library name and version:  **cmake/3.18.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by ledocc** [comment]
> add 3.18.0 to config.yml as well

done

**Comment by madebr** [comment]
@uilianries 
This makes the Nth package having problems with long paths on CI.
The build folder is `C:/J/workspace/onan-center-pull-request_PR-2222/2/pr_2222_2_0_2/.conan/data/cmake/3.18.0/_/_/build/f27dcd2d5e453cb01f9324a959461903ce8116a0/` which is 140 characters long.
[The maximum path on Windows is **260 characters**](https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file#maximum-path-length-limitation).
So more than half of the path is taken by the CI.
This path should/*must* get shortened:
e.g. to `C:/J/workspace/c3i-pr-2222/2/2_0_2/.conan/data/cmake/3.18.0/_/_/build/f27dcd2d5e453cb01f9324a959461903ce8116a0/` (111 characters)

As a comparison, the build folder of cmake on my Windows box is 92 characters.

**Comment by Croydon** [comment]
Ha, you were faster with this one than me 😄 

Could you please also add version `3.16.8`?

**Comment by uilianries** [comment]
@madebr Thanks for reporting! Good catch BTW! @danimtb is checking the short_paths problem, I think this case can be related.

**Comment by ledocc** [comment]
> Ha, you were faster with this one than me smile
> 
> Could you please also add version `3.16.8`?

done

**Comment by Croydon** [comment]
There is no obvious reason why this failed

In the meantime 3.18.1 and 3.17.4 got released

@ledocc Could you please add these versions to conandata.yml and config.yml? This will also start the CI again

File: recipes/cmake/3.x.x/conanfile.py [hunk 1]
```diff
@@ -20,6 +20,7 @@ class CMakeConan(ConanFile):
         "with_openssl": "auto",
     }
 
+    short_paths = True
```
Comment (jgsogo) [inline recipes/cmake/3.x.x/conanfile.py:23]: Is this needed? We prefer to avoid the `short_paths` if it is not strictly needed.

Comment (jgsogo) [reply recipes/cmake/3.x.x/conanfile.py:23]: I see it failed in the build, ok. I'm triggering the job again to see if it works now.

**Comment by jgsogo** [comment]
@Croydon I've fixed this PR with the conflicts from the other one adding `3.18.1`. If we want/need `3.17.4`, we can open a new PR.

> In the meantime 3.18.1 and 3.17.4 got released

**Pull request merged**